### PR TITLE
replace webflux with resttemplate

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**
Signed-off-by: walid-sahnoun <walid.sahnoun@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/explore/server/services/StudyService.java [hunk 1]
```diff
@@ -6,138 +6,123 @@
...
-                .header(HEADER_USER_ID, userId)
-                .retrieve()
-                .onStatus(httpStatus -> httpStatus != HttpStatus.OK, ClientResponse::createException)
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, new HttpEntity<>(getHeaders(userId)), Void.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.OK != e.getStatusCode()) {
```
Comment (archendel) [inline src/main/java/org/gridsuite/explore/server/services/StudyService.java:104]: this check will always be true, also, use REMOTE_ERROR

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 2]
```diff
@@ -152,9 +150,15 @@ public MockResponse dispatch(RecordedRequest request) {
                 } else if (path.matches("/v1/elements/" + CONTINGENCY_LIST_UUID) && "GET".equals(request.getMethod())) {
                     return new MockResponse().setBody(formContingencyListAttributesAsString).setResponseCode(200)
                             .addHeader("Content-Type", "application/json; charset=utf-8");
+                } else if (path.contains("/v1/elements/" + CONTINGENCY_LIST_UUID + "/notification?type=UPDATE_DIRECTORY")) {
```
Comment (archendel) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:153]: why do we need this ?

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 3]
```diff
@@ -152,9 +150,15 @@ public MockResponse dispatch(RecordedRequest request) {
...
+                    return new MockResponse().setBody(formContingencyListAttributesAsString).setResponseCode(200)
+                            .addHeader("Content-Type", "application/json; charset=utf-8");
                 } else if (path.matches("/v1/elements/" + FILTER_UUID) && "GET".equals(request.getMethod())) {
                     return new MockResponse().setBody(filterAttributesAsString).setResponseCode(200)
                             .addHeader("Content-Type", "application/json; charset=utf-8");
+                } else if (path.equals("/v1/elements/" + FILTER_UUID + "/notification?type=UPDATE_DIRECTORY")) {
```
Comment (archendel) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:159]: why do we need this ?

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 4]
```diff
@@ -219,208 +223,186 @@ public MockResponse dispatch(RecordedRequest request) {
...
     @Test
-    public void testCreateCase() throws IOException {
+    public void testCreateCase() throws Exception {
         try (InputStream is = new FileInputStream(ResourceUtils.getFile("classpath:" + TEST_FILE))) {
             MockMultipartFile mockFile = new MockMultipartFile("caseFile", TEST_FILE, "text/xml", is);
             MultipartBodyBuilder bodyBuilder = new MultipartBodyBuilder();
             bodyBuilder.part("caseFile", mockFile.getBytes())
-                .filename(TEST_FILE)
-                .contentType(MediaType.TEXT_XML);
+                    .filename(TEST_FILE)
```
Comment (archendel) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:272]: unnecessary change

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 5]
```diff
@@ -219,208 +223,186 @@ public MockResponse dispatch(RecordedRequest request) {
...
     }
 
     @Test
-    public void testCaseCreationError() throws IOException {
+    public void testCaseCreationError() throws Exception {
         try (InputStream is = new FileInputStream(ResourceUtils.getFile("classpath:" + TEST_FILE_WITH_ERRORS))) {
             MockMultipartFile mockFile = new MockMultipartFile("caseFile", TEST_FILE_WITH_ERRORS, "text/xml", is);
             MultipartBodyBuilder bodyBuilder = new MultipartBodyBuilder();
             bodyBuilder.part("caseFile", mockFile.getBytes())
-                .filename(TEST_FILE_WITH_ERRORS)
```
Comment (archendel) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:294]: unnecessary change

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java [hunk 6]
```diff
@@ -219,208 +223,186 @@ public MockResponse dispatch(RecordedRequest request) {
...
-        webTestClient.post()
-                .uri("/v1/explore/script-contingency-lists/{listName}?&parentDirectoryUuid={parentDirectoryUuid}&description={description}}",
-                        "contingencyListScriptName", PARENT_DIRECTORY_UUID, null)
+    public void testCreateScriptContingencyList() throws Exception {
+
+        mockMvc.perform(post("/v1/explore/script-contingency-lists/{listName}?&parentDirectoryUuid={parentDirectoryUuid}&description={description}}",
+                "contingencyListScriptName", PARENT_DIRECTORY_UUID, null)
                 .header("userId", USER1)
                 .contentType(MediaType.APPLICATION_JSON)
-                .body(BodyInserters.fromValue("Contingency list content"))
```
Comment (archendel) [inline src/test/java/org/gridsuite/explore/server/ExploreTest.java:335]: body request has been removed ?

**Review by archendel — CHANGES_REQUESTED** [review]
Some minor changes, mostly around exception handling

File: src/main/java/org/gridsuite/explore/server/services/CaseService.java [hunk 7]
```diff
@@ -10,93 +10,96 @@
...
-            .publishOn(Schedulers.boundedElastic())
-            .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+                .buildAndExpand(id)
+                .toUriString();
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HEADER_USER_ID, userId);
+        try {
+            restTemplate.exchange(caseServerBaseUri + path, HttpMethod.DELETE, new HttpEntity<>(headers), Void.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/CaseService.java:98]: the behaviour was like this 
return webClient.delete()
            .uri(caseServerBaseUri + path)
            .header(HEADER_USER_ID, userId)
            .retrieve()
            .onStatus(httpStatus -> httpStatus != HttpStatus.OK, ClientResponse::createException)
            .bodyToMono(Void.class)
            .publishOn(Schedulers.boundedElastic())
            .log(ROOT_CATEGORY_REACTOR, Level.FINE);
            
            so, i will change the condition like this if ( ! HttpStatus.OK.equals(e.getStatusCode()))

File: src/main/java/org/gridsuite/explore/server/dto/ElementAttributes.java [hunk 8]
```diff
@@ -6,7 +6,10 @@
  */
 package org.gridsuite.explore.server.dto;
 
-import lombok.*;
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/dto/ElementAttributes.java:9]: done

File: src/main/java/org/gridsuite/explore/server/services/ContingencyListService.java [hunk 9]
```diff
@@ -6,160 +6,121 @@
...
-            .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+                .buildAndExpand(id)
+                .toUriString();
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HEADER_USER_ID, userId);
+        try {
+            restTemplate.exchange(actionsServerBaseUri + path, HttpMethod.DELETE, new HttpEntity<>(headers), Void.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.OK != e.getStatusCode()) {
+                throw new ExploreException(DELETE_ACTION_SERVER_FAILED);
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/ContingencyListService.java:57]: the behaviour was like this
return webClient.delete()
            .uri(actionsServerBaseUri + path)
            .header(HEADER_USER_ID, userId)
            .retrieve()
            .onStatus(httpStatus -> httpStatus != HttpStatus.OK, ClientResponse::createException)
            .bodyToMono(Void.class)
            .publishOn(Schedulers.boundedElastic())
            .log(ROOT_CATEGORY_REACTOR, Level.FINE);
            
            so i will change only REMOTE_ERROR

File: src/main/java/org/gridsuite/explore/server/services/DirectoryService.java [hunk 10]
```diff
@@ -43,138 +42,138 @@ public class DirectoryService implements IDirectoryElementsService {
...
+                .fromPath(ELEMENTS_SERVER_ROOT_PATH + "/{elementUuid}/notification?type={update_directory}")
+                .buildAndExpand(elementUuid, NotificationType.UPDATE_DIRECTORY.name())
+                .toUriString();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HEADER_USER_ID, userId);
+        try {
+            restTemplate.exchange(directoryServerBaseUri + path, HttpMethod.POST, new HttpEntity<>(headers), Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(NOTIFICATION_DIRECTORY_CHANGED);
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/DirectoryService.java:122]: ok

File: src/main/java/org/gridsuite/explore/server/services/DirectoryService.java [hunk 11]
```diff
@@ -43,138 +42,138 @@ public class DirectoryService implements IDirectoryElementsService {
...
     }
 
     // TODO get id/type recursively then do batch delete
     @Override
-    public Mono<Void> delete(UUID id, String userId) {
-        return getDirectoryElements(id, userId).flatMap(e -> deleteElement(e.getElementUuid(), userId))
-            .then();
+    public void delete(UUID id, String userId) {
+        List<ElementAttributes> elementAttributesList = getDirectoryElements(id, userId);
+        if (elementAttributesList != null) {
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/DirectoryService.java:172]: ok i will remove the check

File: src/main/java/org/gridsuite/explore/server/ExploreException.java [hunk 12]
```diff
@@ -14,12 +14,17 @@
 public class ExploreException extends RuntimeException {
 
     public enum Type {
-        STUDY_NOT_FOUND,
-        CONTINGENCY_LIST_NOT_FOUND,
         FILTER_NOT_FOUND,
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/explore/server/ExploreException.java:17]: why removing STUDY_NOT_FOUND, CONTINGENCY_LIST_NOT_FOUND and keeping FILTER_NOT_FOUND ?
(only rest request error to filter server is checked to generate FILTER_NOT_FOUND exception)

File: src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java [hunk 13]
```diff
@@ -30,18 +29,16 @@ protected ResponseEntity<Object> handleException(RuntimeException exception) {
         }
         ExploreException exploreException = (ExploreException) exception;
         switch (exploreException.getType()) {
-            case STUDY_NOT_FOUND:
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java:33]: See remark above on ExploreException

File: src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java [hunk 14]
```diff
@@ -30,18 +29,16 @@ protected ResponseEntity<Object> handleException(RuntimeException exception) {
...
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body(exploreException.getType());
             case NOT_ALLOWED:
                 return ResponseEntity.status(HttpStatus.FORBIDDEN).body(NOT_ALLOWED);
             case REMOTE_ERROR:
                 return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(exception.getMessage());
             case UNKNOWN_ELEMENT_TYPE:
-                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(UNKNOWN_ELEMENT_TYPE);
+            case INSERT_STUDY_FAILED:
+            case NOTIFICATION_DIRECTORY_CHANGED:
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(exploreException.getType());
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java:41]: why adding exception type in the body for some exception types, and not for other ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/explore/server/ExploreException.java (hunk 12 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/ExploreException.java:17]: well the STUDY_NOT_FOUND, CONTINGENCY_LIST_NOT_FOUND where declared but not used and FILTER_NOT_FOUND was used

File: src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java (hunk 14 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java:41]: i will keep the same file i will only remove STUDY_NOT_FOUND, CONTINGENCY_LIST_NOT_FOUND because they are not used

File: src/main/java/org/gridsuite/explore/server/services/StudyService.java (hunk 1 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/StudyService.java:106]: the behaviour was like this
return webClient.delete()
                .uri(studyServerBaseUri + path)
                .header(HEADER_USER_ID, userId)
                .retrieve()
                .onStatus(httpStatus -> httpStatus != HttpStatus.OK, ClientResponse::createException)
                .bodyToMono(Void.class)
                .publishOn(Schedulers.boundedElastic())
                .log(ROOT_CATEGORY_REACTOR, Level.FINE);

i will change HttpMethod.POST with HttpMethod.DELETE

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 15]
```diff
@@ -54,148 +54,137 @@ public ExploreService(
...
-            }
-            return filterService.replaceFilterWithScript(id).doOnSuccess(unused ->
-                directoryService.notifyDirectoryChanged(id, userId).subscribe());
-        });
-    }
-
-    public Mono<Void> deleteElement(UUID id, String userId) {
-        return directoryService.deleteElement(id, userId)
-                .doOnSuccess(e -> directoryService.deleteDirectoryElement(id, userId).subscribe())
-                // FIXME dirty fix to ignore errors and still delete the elements in the directory-server. To delete when handled properly.
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:195]: ok

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 16]
```diff
@@ -54,148 +54,137 @@ public ExploreService(
         this.caseService = caseService;
     }
 
-    public Mono<Void> createStudy(String studyName, UUID caseUuid, String description, String userId, UUID parentDirectoryUuid, Map<String, Object> importParams) {
+    public void createStudy(String studyName, UUID caseUuid, String description, String userId, UUID parentDirectoryUuid, Map<String, Object> importParams) {
         ElementAttributes elementAttributes = new ElementAttributes(UUID.randomUUID(), studyName, STUDY, null, userId, 0L, description);
-
-        return studyService.insertStudyWithExistingCaseFile(elementAttributes.getElementUuid(), userId, caseUuid, importParams)
-                .doOnSuccess(unused -> directoryService.createElement(elementAttributes, parentDirectoryUuid, userId).subscribe());
+        try {
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java]: ok i will use the try/catch in the restTemplate call

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 17]
```diff
@@ -31,21 +33,19 @@ public class ExploreService {
...
     private StudyService studyService;
     private ContingencyListService contingencyListService;
     private FilterService filterService;
     private CaseService caseService;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ExploreService.class);
-
     public ExploreService(
-        DirectoryService directoryService,
-        StudyService studyService,
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:45]: ok, i will fixe it

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 18]
```diff
@@ -31,21 +33,19 @@ public class ExploreService {
     static final String CONTINGENCY_LIST = "CONTINGENCY_LIST";
     static final String FILTER = "FILTER";
     static final String DIRECTORY = "DIRECTORY";
-
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExploreService.class);
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:34]: i use the logger

File: pom.xml [hunk 19]
```diff
@@ -187,5 +187,9 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-web</artifactId>
         </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply pom.xml]: done

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 2 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java]: we need for a specific test

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 3 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:153]: we need it for a specific test

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 4 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:267]: ok, i fix it

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 5 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:294]: ok

File: src/test/java/org/gridsuite/explore/server/ExploreTest.java (hunk 6 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/explore/server/ExploreTest.java:335]: i used again

File: pom.xml (hunk 19 above)
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply pom.xml]: done

File: src/main/java/org/gridsuite/explore/server/ExploreController.java [hunk 20]
```diff
@@ -46,173 +45,189 @@ public ExploreController(ExploreService exploreService, DirectoryService directo
...
-    public ResponseEntity<Mono<Void>> deleteElement(@PathVariable("elementUuid") UUID elementUuid,
-                                                    @RequestHeader("userId") String userId) {
-        return ResponseEntity.ok().body(exploreService.deleteElement(elementUuid, userId));
+    public ResponseEntity<Void> deleteElement(@PathVariable("elementUuid") UUID elementUuid,
+                                              @RequestHeader("userId") String userId) {
+        exploreService.deleteElement(elementUuid, userId);
+        return ResponseEntity.ok().build();
     }
 
     @GetMapping(value = "/explore/elements/metadata")
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/ExploreController.java]: ok

File: src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java (hunk 13 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/RestResponseEntityExceptionHandler.java:33]: ok

File: src/main/java/org/gridsuite/explore/server/services/FilterService.java [hunk 21]
```diff
@@ -6,139 +6,94 @@
...
                 .toUriString();
-
-        return webClient.put()
-                .uri(filterServerBaseUri + path)
-                .retrieve()
-                .onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new ExploreException(FILTER_NOT_FOUND)))
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        restTemplate.exchange(filterServerBaseUri + path, HttpMethod.PUT, null, Void.class);
```
Comment (archendel) [inline src/main/java/org/gridsuite/explore/server/services/FilterService.java:49]: keep the following behaviour 
.onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new ExploreException(FILTER_NOT_FOUND)))

File: src/main/java/org/gridsuite/explore/server/services/FilterService.java [hunk 22]
```diff
@@ -6,139 +6,94 @@
...
-        return webClient.post()
-                .uri(filterServerBaseUri + path)
-                .retrieve()
-                .onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new ExploreException(FILTER_NOT_FOUND)))
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+                .buildAndExpand(id, newId)
+                .toUriString();
+        restTemplate.exchange(filterServerBaseUri + path, HttpMethod.POST, null, Void.class);
```
Comment (archendel) [inline src/main/java/org/gridsuite/explore/server/services/FilterService.java:56]: keep the following behaviour :
.onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new ExploreException(FILTER_NOT_FOUND)))

File: src/main/java/org/gridsuite/explore/server/services/ContingencyListService.java [hunk 23]
```diff
@@ -9,157 +9,105 @@
...
-
-        return webClient.post()
-            .uri(actionsServerBaseUri + path)
-            .retrieve()
-            .bodyToMono(Void.class)
-            .publishOn(Schedulers.boundedElastic())
-            .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+                .buildAndExpand(id, newId)
+                .toUriString();
+        HttpHeaders headers = new HttpHeaders();
```
Comment (archendel) [inline src/main/java/org/gridsuite/explore/server/services/ContingencyListService.java:91]: this header was not here before, do we need it ?

File: src/main/java/org/gridsuite/explore/server/services/StudyService.java [hunk 24]
```diff
@@ -6,138 +6,117 @@
...
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.add(HEADER_USER_ID, userId);
+        HttpEntity<Map<String, Object>> request = new HttpEntity<>(
+                importParams, headers);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, request, Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(INSERT_STUDY_FAILED);
```
Comment (archendel) [inline src/main/java/org/gridsuite/explore/server/services/StudyService.java:62]: remove this try catch that was not here before to handle the exception with the exception handler ?

File: src/main/java/org/gridsuite/explore/server/services/StudyService.java [hunk 25]
```diff
@@ -6,138 +6,117 @@
...
+        }
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.MULTIPART_FORM_DATA);
+        headers.add(HEADER_USER_ID, userId);
+        HttpEntity<MultiValueMap<String, HttpEntity<?>>> request = new HttpEntity<>(
+                multipartBodyBuilder.build(), headers);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, request, Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(INSERT_STUDY_FAILED);
```
Comment (archendel) [inline src/main/java/org/gridsuite/explore/server/services/StudyService.java:85]: remove this try catch that was not here before to handle the exception with the exception handler ?

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java (hunk 18 above)
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:34]: Sure, but you removed it at the line 41 to add it here

**Review by archendel — CHANGES_REQUESTED** [review]
some try catches have been removed

File: src/main/java/org/gridsuite/explore/server/services/StudyService.java (hunk 25 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/StudyService.java]: done

File: src/main/java/org/gridsuite/explore/server/services/StudyService.java (hunk 24 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/StudyService.java]: done

File: src/main/java/org/gridsuite/explore/server/services/IDirectoryElementsService.java [hunk 26]
```diff
@@ -23,23 +21,25 @@
...
-                    return elementAttribute;
-                }));
+        List<Map<String, Object>> metadata = getMetadata(lstElementAttribute.stream().map(ElementAttributes::getElementUuid).collect(Collectors.toList()));
+        return metadata.stream().map(metadataItem ->
+                populateMedataItem(mapElementAttribute.get(metadataItem.getOrDefault("id", "").toString()), metadataItem)
+        ).collect(Collectors.toList());
+    }
+
+    private ElementAttributes populateMedataItem(ElementAttributes elementAttributes, Map<String, Object> metadataItem) {
+        elementAttributes.setSpecificMetadata(metadataItem);
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/IDirectoryElementsService.java:42]: i add exception handler for all exceptions

File: src/main/java/org/gridsuite/explore/server/services/FilterService.java (hunk 21 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/FilterService.java]: ok i will return the try catch bloc here

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java (hunk 18 above)
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java]: ok it's done

File: src/main/java/org/gridsuite/explore/server/ExploreException.java (hunk 12 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/explore/server/ExploreException.java:17]: Yes, but study and contingency could be treated the same way as filters ... (catching exception)

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/explore/server/ExploreException.java (hunk 12 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/ExploreException.java:17]: yes, but the old behaviour does not define how to treat them,
for me i add an exception handler to treat them globally.
should i keep them like the old behaviour or not ?

File: src/main/java/org/gridsuite/explore/server/ExploreController.java [hunk 27]
```diff
@@ -46,173 +45,189 @@ public ExploreController(ExploreService exploreService, DirectoryService directo
...
+                                           @RequestParam(QUERY_PARAM_PARENT_DIRECTORY_ID) UUID parentDirectoryUuid,
+                                           @RequestHeader("userId") String userId) {
+        exploreService.createCase(caseName, caseFile, description, userId, parentDirectoryUuid);
+        return ResponseEntity.ok().build();
     }
 
     @PostMapping(value = "/explore/cases")
     @Operation(summary = "Duplicate a case")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Case duplication request delegated to case server")})
     public ResponseEntity<Mono<Void>> createCase(@RequestParam("duplicateFrom") UUID parentCaseUuid,
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/explore/server/ExploreController.java]: done

**Review by archendel — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]
Review: OK
Tests: OK
